File: ProductManagementSystem.Domain/Entities/Category.cs
using ProductManagementSystem.Domain.Common;
using System.Text.Json.Serialization;

namespace ProductManagementSystem.Domain.Entities;

public class Category : BaseEntity
{
    public string NameEn { get; set; }
    public string NameAr { get; set; }
    [JsonIgnore] // Add this attribute to avoid circular reference
    public IEnumerable<Product> Products { get; set; }
}

File: productmanagementsystem.client/src/models/entities/category.ts
import { IProductDTO } from './product';

export interface ICategoryDTO {
  id: string;
  nameEn: string;
  nameAr: string;
  products: IProductDTO[];
}

export interface CreateOrUpdateCategoryFormData {
  nameEn: string;
  nameAr: string;
}

export interface CreateCategoryRequest {
  nameEn: string;
  nameAr: string;
}

export interface UpdateCategoryRequest {
  id: string;
  nameEn: string;
  nameAr: string;
}
